### PR TITLE
Fix location of node binary on Windows

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: defs.bzl [hunk 1]
```diff
@@ -16,5 +16,5 @@
 
 Users should not load files under "/internal"
 """
-load("//internal:node.bzl", "nodejs_binary")
 load("//internal:node_install.bzl", "node_repositories")
+load("//internal:node.bzl", "nodejs_binary")
```
Comment (mprobst) [inline defs.bzl]: is this change intentional?

File: internal/node_install.bzl [hunk 2]
```diff
@@ -48,6 +41,8 @@ alias(name = "npm", actual = "bin/npm")
         stripPrefix = "node-v6.10.2-win-x64",
         sha256 = "d778ed84685c6604192cfcf40192004e27fb11c9e65c3ce4b283d90703b4192c"
     )
+    node = "node.exe"
```
Comment (mprobst) [inline internal/node_install.bzl]: why is this `bin/node` above, but just `node.exe` here? Add a comment?

File: internal/node_install.bzl [hunk 3]
```diff
@@ -86,14 +90,18 @@ package(default_visibility = ["//visibility:public"])
 exports_files(['yarn.sh'])
 alias(name = "yarn", actual = ":yarn.sh")
 """)
+  if ctx.os.name.lower().find("windows") != -1:
+    node = Label("@nodejs//:node.exe")
```
Comment (mprobst) [inline internal/node_install.bzl]: why are you not using the alias defined above, but special case node.exe for windows?

File: defs.bzl (hunk 1 above)
Comment (alexeagle) [reply defs.bzl]: just changing the order to be more intuitive, but it has no effect on the PR, I'll remove for clarity

File: internal/node_install.bzl (hunk 2 above)
Comment (alexeagle) [reply internal/node_install.bzl]: adding comment

File: internal/node_install.bzl (hunk 3 above)
Comment (alexeagle) [reply internal/node_install.bzl]: the alias isn't resolved in skylark, if you `ctx.path(Label(alias))` you get the location of the alias not the target.
```
Not a file: /usr/local/google/home/alexeagle/.cache/bazel/_bazel_alexeagle/c410d6070ef2d359d26ac4aeb886e1ea/external/nodejs/node
```
I guess the alias resolution happens in the loading phase?

Comment (mprobst) [reply internal/node_install.bzl]: Maybe something to task? Also, why do you have the alias then in the first place?
